flask_user/forms.py

```diff
@@ -210,7 +221,7 @@ def validate(self):
             user, user_email = user_manager.find_user_by_email(self.email.data)
 
         # Handle successful authentication
-        if user and user.password and user_manager.verify_password(self.password.data, user):
+        if user and user_manager.verify_password(self.password.data, user):
```

Comment:
For heterogeneous authorization systems, some users do not have a local `password` (such as those users relying on an external OAuth server).  Without a defined password, an `attribute error` is raised when trying to pass `self.password.data` into `verify_password` - thus mandating the extra check.